library/src/main/java/org/apache/apex/malhar/lib/utils/serde/SerToSerializeBuffer.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.apex.malhar.lib.utils.serde;
+
+import com.datatorrent.netlet.util.Slice;
+
+public interface SerToSerializeBuffer<T> extends Serde<T, Slice>, ResetableWindowListener
```

Comment:
Also, I see you have instanceof SerToSerializeBuffer everywhere. I think since the interface Serde is only used in Spillable DS right now, it's okay to change the existing Serde interface to include the buffer related methods, as we know that without the use of a buffer, serialization is going to be inefficient.